cli/command/formatter/displayutils.go

```diff
@@ -0,0 +1,82 @@
+package formatter
+
+import (
+	"unicode/utf8"
+
+	"golang.org/x/text/width"
+)
+
+// charWidth returns the number of horizontal positions a character occupies,
+// and is used to account for wide characters when displaying strings.
+//
+// In a broad sense, wide characters include East Asian Wide, East Asian Full-width,
+// (when not in East Asian context) see http://unicode.org/reports/tr11/.
+func charWidth(r rune) int {
+	switch width.LookupRune(r).Kind() {
+	case width.EastAsianWide, width.EastAsianFullwidth:
+		return 2
+	default:
+		return 1
+	}
+}
+
+// TruncateByWidth truncates a string to fit within maxwidth.
+// For maxwidth of 1, first char of string will return even if its width > 1.
+func TruncateByWidth(s string, maxwidth int) string {
+	if maxwidth <= 0 {
+		return ""
+	}
+	byteLen := len(s)
+	if byteLen == utf8.RuneCountInString(s) {
+		r := []rune(s)
+		if byteLen <= maxwidth {
+			return s
+		}
+		return string(r[:maxwidth])
+	}
+
+	var (
+		display      []int
+		displayWidth int
+	)
+	rs := []rune(s)
+	for _, r := range rs {
+		cw := charWidth(r)
+		displayWidth += cw
+		display = append(display, displayWidth)
+	}
+	if displayWidth <= maxwidth {
+		return s
+	}
+	if display[0] >= maxwidth {
+		return string(rs[0])
+	}
+	for i := range display {
+		if display[i] <= maxwidth && display[i+1] > maxwidth {
+			s = string(rs[:i+1])
+		}
+	}
+	return s
+}
+
+// Ellipsis truncates a string to fit within maxwidth, and appends ellipsis (…).
+// For maxwidth of 1 and lower, no ellipsis is appended.
+func Ellipsis(s string, maxwidth int) string {
+	if maxwidth <= 0 {
+		return ""
+	}
+	if maxwidth == 1 {
+		return TruncateByWidth(s, 1)
+	}
+
+	displayWidth := 0
+	rs := []rune(s)
+	for _, r := range rs {
+		cw := charWidth(r)
+		displayWidth += cw
+	}
```

Comment:
It seems like we know at this point if the `displayWidth` is greater than or equal to the maxWidth. Could we actually return here instead of calling `TruncateByWidth()` ?

```go
for index, r := range rs {
    width := charWidth(r)
    switch {
    case displayWidth + width == maxWidth && index + 1 == len(rs):
        return s
    case displayWidth + width >= maxWidth:
        return string(rs[:index]) + "…"
    }
    displayWidth += width
}
return s
```

If this works, then I think we can simplify the `maxwidth == 1` case as well and we could remove `TruncateByWidth()` entirely.